packages/site/src/hooks/MetamaskContext.tsx

```diff
@@ -0,0 +1,124 @@
+import {
+  createContext,
+  Dispatch,
+  ReactNode,
+  Reducer,
+  useEffect,
+  useReducer,
+} from 'react';
+import { isFlask, isSnapInstalled } from '../utils';
+
+export type MetamaskState = {
+  isSnapInstalled: boolean;
+  isFlask: boolean;
+  error?: Error;
+};
+
+const initialState: MetamaskState = {
+  isSnapInstalled: false,
+  isFlask: false,
+  error: undefined,
+};
+
+type MetamaskDispatch = { type: MetamaskActions; payload: any };
+
+export const MetaMaskContext = createContext<
+  [MetamaskState, Dispatch<MetamaskDispatch>]
+>([
+  initialState,
+  () => {
+    /* no op */
+  },
+]);
+
+export enum MetamaskActions {
+  SetInstalled = 'SetInstalled',
+  SetFlaskDetected = 'SetFlaskDetected',
+  SetError = 'SetError',
+}
+
+const reducer: Reducer<MetamaskState, MetamaskDispatch> = (state, action) => {
+  switch (action.type) {
+    case MetamaskActions.SetInstalled:
+      return {
+        ...state,
+        isSnapInstalled: action.payload,
+      };
+
+    case MetamaskActions.SetFlaskDetected:
+      return {
+        ...state,
+        isFlask: action.payload,
+      };
+
+    case MetamaskActions.SetError:
+      return {
+        ...state,
+        error: action.payload,
+      };
+
+    default:
+      return state;
+  }
+};
+
+/**
+ * MetaMask context provider to handle MetaMask and snap status.
+ *
+ * @param props - React Props.
+ * @param props.children - React component to be wrapped by the Provider.
+ * @returns JSX.
+ */
+export const MetaMaskProvider = ({ children }: { children: ReactNode }) => {
+  const [state, dispatch] = useReducer(reducer, initialState);
+
+  useEffect(() => {
+    async function detectFlask() {
+      const isFlaskDetected = await isFlask();
+
+      dispatch({
+        type: MetamaskActions.SetFlaskDetected,
+        payload: isFlaskDetected,
+      });
+    }
+
+    async function detectSnapInstalled() {
+      const snapInstalled = await isSnapInstalled();
+      dispatch({
+        type: MetamaskActions.SetInstalled,
+        payload: snapInstalled,
+      });
+    }
+
+    detectFlask();
+
+    if (state.isFlask) {
+      detectSnapInstalled();
+    }
+  }, [state.isFlask]);
```

Comment:
```suggestion
  }, [state.isFlask, window.ethereum]);
```
That seems to fix my problems with not detecting Flask.